firmware/dev/f4dev/CMakeLists.txt

```diff
@@ -0,0 +1,110 @@
+set(ARM_CORE "cm4")
+set(IOC_PATH "${CMAKE_CURRENT_SOURCE_DIR}/src/cubemx/f4dev.ioc")
+set(SHARED_DIR "${FIRMWARE_DIR}/shared")
+set(LINKER_SCRIPT "${THIRD_PARTY_DIR}/linker/STM32F412RGTX_FLASH.ld")
+
+file(GLOB_RECURSE APP_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/src/app/*.c")
+list(APPEND APP_SRCS "${SHARED_APP_SRCS}")
+set(APP_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/src/app" "${SHARED_APP_INCLUDE_DIR}")
+
+# file(GLOB_RECURSE IO_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/src/io/*.c")
+# list(APPEND IO_SRCS "${SHARED_IO_SRCS}")
+# list(REMOVE_ITEM IO_SRCS "${SHARED_IO_INCLUDE_DIR}/Io_SharedSpi.c")
+set(IO_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/src/io" "${SHARED_IO_INCLUDE_DIR}")
+
+file(GLOB_RECURSE HW_SRCS 
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/hw/*.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/cubemx/Src/*.c"
+)
+
+list (APPEND HW_SRCS "${SHARED_DIR}/src/hw/hw_sdio_sd.c")
+# list(APPEND IO_SRCS "${SHARED_HW_SRCS}")
+set(HW_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/src/hw" "${SHARED_HW_INCLUDE_DIR}")
+
+# file(GLOB_RECURSE TEST_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/Test/Src/*.cpp")
+# set(TEST_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/Test/Inc")
+# list(APPEND TEST_INCLUDE_DIRS "${SHARED_TEST_UTILS_INCLUDE_DIRS}")
+
+# Autogenerate jsoncan library.
+# jsoncan_library(
+#     "f4dev_can"
+#     "f4dev"
+#     "${CMAKE_CURRENT_BINARY_DIR}"
+#     "${ARM_CORE}"
+# )
```

Comment:
I'd rather you just delete this stuff rather than comment it out. Here and the tests below.